src/redteam/extraction/base.ts

```diff
@@ -0,0 +1,67 @@
+import dedent from 'dedent';
+import logger from '../../logger';
+import { ApiProvider } from '../../types';
+
+/**
+ * Abstract base class for extraction operations.
+ * @template T The type of data to be extracted.
+ */
+export abstract class ExtractionBase<T> {
```

Comment:
I'm skeptical of this abstraction.  It seems like langchain-style indirection around a single `callApi`.  The prompt construction and output parsing is unrelated between purpose and entities.  How much do these extractors actually have in common?

Reply:
You're absolutely right. I'll simplify them / remove them entirely.